app/components/citySearch.js

```diff
@@ -0,0 +1,33 @@
+'use client';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { fetchWeather } from '../store/slices/cities';
+
+export default function CitySearch() {
+	//setting up useState and Dispatch
+	const dispatch = useDispatch();
+	const [city, setCity] = useState('');
+	//submit button function that requests API and formats API and sets useState back to ''
+	const handleFormSubmit = (event) => {
+		event.preventDefault();
+		dispatch(fetchWeather(city));
+		setCity('');
```

Comment:
its a better UX to clear the input only after the call is successful. If the user enters a bad input, you want to see it and not clear it